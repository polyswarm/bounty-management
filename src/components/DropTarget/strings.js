const strings = {
  dragAndDrop: 'Drag and drop your files here, or use the button below.'
}
export default strings;
