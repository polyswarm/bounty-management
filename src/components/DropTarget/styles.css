@keyframes lighten {
  0% { background-image: url('/img/gray_hexgrid_2x.png'); }
  100% { background-image: url('/img/hexgrid_2x.png'); }
}

@keyframes darken {
  0% { background-image: url('/img/hexgrid_2x.png'); }
  100% { background-image: url('/img/gray_hexgrid_2x.png'); }
}

.Drop-Target {
  animation-duration: 1s;
  animation-name: darken;
  background-color: #fafafa;
  background-image: url('/img/gray_hexgrid_2x.png');
  background-size: 48px 58px;
  border: 2px dashed black;
  border-radius: 20px;
  color: black;
  cursor: pointer;
  display: table;
  height: 100%;
  min-width: 350px;
  position: relative;
  vertical-align: middle;
}

.Drop-Target > div {
  display: table-cell;
  padding: 0 20px;
  vertical-align: middle;
}

p {
  display: inline-block;
  margin: 5px auto;
}

.drag {
  animation-duration: 1s;
  animation-name: lighten;
  background-image: url('/img/hexgrid_2x.png');
  cursor: copy;
}
