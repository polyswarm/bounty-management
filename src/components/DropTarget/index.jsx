// Vendor imports
import React, {Component} from 'react';
// Bounty imports
import FileButton from '../FileButton';
// Component imports
import strings from './strings';
import './styles.css';

class DropTarget extends Component {
  constructor(props) {
    super(props);
    this.onDropHandler = this.onDropHandler.bind(this);
    this.onDragOverHandler = this.onDragOverHandler.bind(this);
    this.onDragEnterHandler = this.onDragEnterHandler.bind(this);
    this.onDragLeaveHandler = this.onDragLeaveHandler.bind(this);
    this.onFileSelectedHandler = this.onFileSelectedHandler.bind(this);
  }

  render() {
    return(
      <div className='Drop-Target'
        ref={(target) => {this.target = target;}}
        onDrop={this.onDropHandler}
        onDragOver={this.onDragOverHandler}
        onDragEnter={this.onDragEnterHandler}
        onDragLeave={this.onDragLeaveHandler}>
        <div>
          <p>{strings.dragAndDrop}</p>
          <FileButton onFileSelected={this.onFileSelectedHandler}/>
        </div>
      </div>
    );
  }

  onDropHandler(event) {
    const { props: { onFilesSelected } } = this;
    // Don't want file to open
    event.preventDefault();

    let files = [];

    if (onFilesSelected) {
      if (event.dataTransfer.items) {
        const items = event.dataTransfer.items;
        for (var i = 0; i < items.length; i++) {
          const item = items[i];
          if (item.kind === 'file') {
            files.push(item.getAsFile());
          }
        }
      } else {
        files = event.dataTransfer.files;
      }

      if (files.length > 0) {
        onFilesSelected(files);
      }
    }

    this.removeDragData(event);
    this.target.classList.remove('drag');
  }

  onDragOverHandler(event) {
    event.preventDefault();
  }

  onDragEnterHandler(event) {
    event.preventDefault();
    this.target.classList.add('drag');
  }

  onDragLeaveHandler(event) {
    event.preventDefault();
    this.target.classList.remove('drag');
  }

  onFileSelectedHandler(file) {
    const { props: { onFilesSelected } } = this;
    if (onFilesSelected) {
      onFilesSelected([file]);
    }
  }

  removeDragData(event) {
    if (event.dataTransfer.items) {
      event.dataTransfer.items.clear();
    } else {
      event.dataTransfer.clearData();
    }
  }
}
export default DropTarget;
