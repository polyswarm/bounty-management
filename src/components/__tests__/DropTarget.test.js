import React from 'react';
import Adapter from 'enzyme-adapter-react-16'
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json'
import DropTarget from '../DropTarget';

it('DropTarget renders without crashing', () => {
  const wrapper = render(<DropTarget />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});
