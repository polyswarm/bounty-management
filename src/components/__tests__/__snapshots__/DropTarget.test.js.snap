// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropTarget renders without crashing 1`] = `
<div
  class="Drop-Target"
>
  <p>
    Drag and drop your files here, or use the button below.
  </p>
</div>
`;
